scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -67,6 +68,9 @@ u32 layered_root_tgid = 0;
 u32 empty_layer_ids[MAX_LAYERS];
 u32 nr_empty_layer_ids;
 
+u32 all_gpu_pids[MAX_GPU_PIDS];
+u32 cur_gpu_pids[MAX_GPU_PIDS];
```

Comment:
You probably want a variable to track `nr_gpu_pids` as well so that indexing into the array on the bpf side is possible.

Another thought is it is probably also worth storing the GPU/NUMA node mappings as well on the bpf side. That way on the bpf side we know what resources the pid is using in relation to the topology.